python3/game_state.py

```diff
@@ -105,37 +108,37 @@ def filter_entity_fn(entity):
         self._state["entities"] = list(filter(
             filter_entity_fn, self._state["entities"]))
 
-    def _on_agent_state(self, agent_state):
-        agent_number = agent_state.get("number")
-        self._state["agent_state"][str(agent_number)] = agent_state
+    def _on_unit_state(self, unit_state):
+        unit_id = unit_state.get("unit_id")
+        self._state["unit_state"][unit_id] = unit_state
 
     def _on_entity_state(self, x, y, updated_entity):
         for entity in self._state.get("entities"):
             if entity.get("x") == x and entity.get("y") == y:
                 self._state["entities"].remove(entity)
         self._state["entities"].append(updated_entity)
 
-    def _on_agent_action(self, agent_number, action_packet):
-        agent = self._state["agent_state"][str(agent_number)]
-        coordinates = agent.get("coordinates")
+    def _on_unit_action(self, action_packet):
+        unit_id = action_packet["unit_id"]
+        unit = self._state["unit_state"][unit_id]
+        coordinates = unit.get("coordinates")
         action_type = action_packet.get("type")
         if action_type == "move":
             move = action_packet.get("move")
-            if move in _agent_move_set:
-                new_coordinates = self._get_new_agent_coordinates(
+            if move in _move_set:
+                new_coordinates = self._get_new_unit_coordinates(
                     coordinates, move)
-                self._state["agent_state"][str(
-                    agent_number)]["coordinates"] = new_coordinates
+                self._state["unit_state"][unit_id]["coordinates"] = new_coordinates
         elif action_type == "bomb":
-            # no - op since this is redudnant info
+            # no - op since this is redundant info
             pass
         elif action_type == "detonate":
-            # no - op since this is redudnant info
+            # no - op since this is redundant info
             pass
         else:
             print(f"Unhandled agent action recieved: {action_type}")
 
-    def _get_new_agent_coordinates(self, coordinates, move_action) -> [int, int]:
+    def _get_new_unit_coordinates(self, coordinates, move_action) -> [int, int]:
         [x, y] = coordinates
         if move_action == "up":
```

Comment:
Also need to update the tests https://github.com/CoderOneHQ/starter-kits/blob/6403659c68f93e053f47b3b67f0976c073ee700e/python3/test_game_state.py#L1

and also the json schema (from the game-server repo)